drivers/digital-io/max22190/iio_max22190.c

```diff
@@ -0,0 +1,608 @@
+}
+
+/**
+ * @brief Write the filter bypass attribute for a specific channel
+ *
+ * @param dev - The iio device structure
+ *
+ * @param buf - Buffer to be filled with requested data.
+ *
+ * @param len - Length of the received command buffer in bytes.
+ *
+ * @param channel - Command channel info.
+ *
+ * @param priv - Private descriptor.
+ *
+ * @return 0 in case of succes, error code otherwise.
+*/
+static int max22190_iio_write_filter(void *dev, char *buf, uint32_t len,
+				     const struct iio_ch_info *channel,
+				     intptr_t priv)
+{
+	struct max22190_iio_desc *desc = dev;
+	int ch;
+	uint32_t val;
+
+	ch = channel->ch_num;
+
+	iio_parse_value(buf, IIO_VAL_INT, (int32_t *)&val, NULL);
+
+	if (val > 1)
+		return -EINVAL;
+
+	return max22190_reg_update(desc->max22190_desc,
+				   MAX22190_FILTER_IN_REG(ch),
+				   MAX22190_FBP_MASK, val);
+}
+
+/**
+ * @brief Read the filter delay attribute for a specific channel
+ *
+ * @param dev - The iio device structure
+ *
+ * @param buf - Buffer to be filled with requested data.
+ *
+ * @param len - Length of the received command buffer in bytes.
+ *
+ * @param channel - Command channel info.
+ *
+ * @param priv - Private descriptor.
+ *
+ * @return 0 in case of succes, error code otherwise.
+*/
+static int max22190_iio_read_filter_delay(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+	struct max22190_iio_desc *desc = dev;
+	int ret, ch;
+	uint32_t val;
+
+	ch = channel->ch_num;
+
+	ret = max22190_reg_read(desc->max22190_desc, MAX22190_FILTER_IN_REG(ch),
+				&val);
+	if (ret)
+		return ret;
+
+	val = max22190_delay_avail[no_os_field_get(MAX22190_DELAY_MASK, val)];
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, (int32_t *)&val);
+}
+
+/**
+ * @brief Write the filter delay attribute for a specific channel
+ *
+ * @param dev - The iio device structure
+ *
+ * @param buf - Buffer to be filled with requested data.
+ *
+ * @param len - Length of the received command buffer in bytes.
+ *
+ * @param channel - Command channel info.
+ *
+ * @param priv - Private descriptor.
+ *
+ * @return 0 in case of succes, error code otherwise.
+*/
+static int max22190_iio_write_filter_delay(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+	struct max22190_iio_desc *desc = dev;
+	int ch;
+	uint32_t val, i;
+
+	ch = channel->ch_num;
+
+	iio_parse_value(buf, IIO_VAL_INT, (int32_t *)&val, NULL);
+
+	for (i = 0; i < NO_OS_ARRAY_SIZE(max22190_delay_avail); i++) {
+		if (val == max22190_delay_avail[i])
+			break;
+
+		if (i == NO_OS_ARRAY_SIZE(max22190_delay_avail) - 1)
+			return -EINVAL;
+	}
+
+	return max22190_reg_update(desc->max22190_desc,
+				   MAX22190_FILTER_IN_REG(ch),
+				   MAX22190_DELAY_MASK,
+				   no_os_field_prep(MAX22190_DELAY_MASK, i));
+}
+
+/**
+ * @brief Read the available values for filter delay attribute for a specific
+ * 	  channel
+ *
+ * @param dev - The iio device structure
+ *
```

Comment:
idk about these blank lines... i think we'd be better without

Reply:
Will remove them.